packages/flutter/lib/src/cupertino/context_menu.dart

```diff
@@ -422,81 +433,79 @@ class _DecoyChild extends StatefulWidget {
 
   @override
   void initState() {
     super.initState();
-    // Change the color of the child during the initial part of the decoy bounce
-    // animation. The interval was eyeballed from a physical iOS 13.1.2 device.
-    _mask = _OnOffAnimation<Color>(
-      controller: widget.controller,
-      onValue: _lightModeMaskColor,
-      offValue: _masklessColor,
-      intervalOn: 0.0,
-      intervalOff: 0.5,
-    );
-
-    final Rect midRect =  widget.beginRect!.deflate(
+    final Rect midRect = widget.beginRect!.deflate(
       widget.beginRect!.width * (_kOpenScale - 1.0) / 2,
     );
     _rect = TweenSequence<Rect?>(<TweenSequenceItem<Rect?>>[
       TweenSequenceItem<Rect?>(
         tween: RectTween(
           begin: widget.beginRect,
-          end: midRect,
-        ).chain(CurveTween(curve: Curves.easeInOutCubic)),
-        weight: 1.0,
+          end: widget.beginRect,
+        ).chain(CurveTween(curve: Curves.linear)),
+        weight: 3,
       ),
       TweenSequenceItem<Rect?>(
         tween: RectTween(
-          begin: midRect,
+          begin: widget.beginRect,
           end: widget.endRect,
-        ).chain(CurveTween(curve: Curves.easeOutCubic)),
-        weight: 1.0,
+        ).chain(CurveTween(curve: Curves.easeOutSine)),
+        weight: 5,
       ),
     ]).animate(widget.controller);
-    _rect.addListener(_rectListener);
-  }
 
-  // Listen to the _rect animation and vibrate when it reaches the halfway point
-  // and switches from animating down to up.
-  void _rectListener() {
-    if (widget.controller.value < 0.5) {
-      return;
-    }
-    HapticFeedback.selectionClick();
-    _rect.removeListener(_rectListener);
+    const List<BoxShadow> endBoxShadow = <BoxShadow>[
+      BoxShadow(
+        color: Color(0x40000000),
+        blurRadius: 10.0,
+        spreadRadius: 0.5,
+      )
+    ];
+
+    _boxDecoration = DecorationTween(
+      begin: BoxDecoration(
+          color: const Color(0xFFFFFFFF),
+          borderRadius: widget.child.runtimeType == Container
+              ? ((widget.child as Container?)?.decoration as BoxDecoration?)
+                      ?.borderRadius ??
+                  BorderRadius.circular(0)
+              : BorderRadius.circular(0),
+          boxShadow: widget.child.runtimeType == Container
+              ? ((widget.child as Container?)?.decoration as BoxDecoration?)
+                      ?.boxShadow ??
+                  const <BoxShadow>[]
+              : const <BoxShadow>[]),
+      end: BoxDecoration(
+        color: const Color(0xFFFFFFFF),
+        borderRadius: widget.child.runtimeType == Container
+            ? ((widget.child as Container?)?.decoration as BoxDecoration?)
+                    ?.borderRadius ??
+                BorderRadius.circular(0)
+            : BorderRadius.circular(0),
+        boxShadow: widget.child.runtimeType == Container
+            ? ((widget.child as Container?)?.decoration as BoxDecoration?)
```

Comment:
I think that this approach of inspecting the type of the child is unreliable and not a typical Flutter pattern.  For example, this won't work if the child is a DecoratedBox instead of a Container, or if the Container is wrapped by another widget, etc.  But I do agree that we should figure out a way that things like nested shadows can look right!

I think the priorities here should be 1. do the right thing by default in simple common cases (like when the child has no shadow) and 2. allow devs to do the work to get it right in more complicated cases.  If we remove this type checking, would it be possible for a developer to get a child with a shadow to look correct during the animation?  Maybe by putting the shadow above the CupertinoContextMenu or by using previewBuilder to build a version of the child with no shadow?  If it's not possible, is there a change we can make to help make it possible?

Reply:
Looking at this again, I'm not sure if the direct type checking is necessary, but the type casting is for what I was trying to do. I agree on the priorities. I didn't want to put this in and had it overwrite and conflict with stylings developers have with their widgets, but still working for simple ones. I'll try what you suggest, and I have some other ideas since I worked on this last that I'll play around with too.

Reply:
Inspecting the type ended up being unnecessary, as well as getting the existing shadow. All I needed was the border radius, which I can do without the type checking. If a shadow is given by the developer it will just layer with this effect and looks natural. It will be removed once it's fully expanded. Compared to iOS, if they add a shadow, it is removed on press, then comes back after a delay once the context menu is closed. Which looks pretty odd to me.

Reply:
Oops. I found an issue with not type checking. It breaks on images.